slack-server-integration-common/src/main/java/com/atlassian/plugins/slack/api/webhooks/GenericMessageSlackEvent.java

```diff
@@ -16,142 +16,51 @@
-    "channel": "CF49QEJF4",
-    "event_ts": "1547500911.010000",
-    "channel_type": "channel"
-  },
-  "type": "event_callback",
-  "event_id": "EvFDLP0K7F",
-  "event_time": 1547500911,
-  "authed_users": [
-    "UF36K27BR"
-  ]
-}
-
-{
-  "token": "XXXXXXXXXXXXYYYYYYYYYYYY",
-  "team_id": "TXXXXXXXX",
-  "api_app_id": "AXXXXXXXX",
-  "event": {
-    "type": "message",
-    "deleted_ts": "1547500911.010000",
-    "subtype": "message_deleted",
-    "hidden": true,
-    "channel": "CF49QEJF4",
-    "previous_message": {
-      "type": "message",
-      "user": "UC7LU9ZGS",
-      "text": "Any message",
-      "client_msg_id": "853ce7b3-b7ff-4096-b202-c3bbf65c1e4a",
-      "ts": "1547500911.010000"
-    },
-    "event_ts": "1547500951.010100",
-    "ts": "1547500951.010100",
-    "channel_type": "channel"
-  },
-  "type": "event_callback",
-  "event_id": "EvFDLPHL69",
-  "event_time": 1547500951,
-  "authed_users": [
-    "UF36K27BR"
-  ]
-}
-
-{
-  "token": "XXXXXXXXXXXXYYYYYYYYYYYY",
-  "team_id": "TXXXXXXXX",
-  "api_app_id": "AXXXXXXXX",
-  "event": {
-    "type": "message",
-    "message": {
-      "type": "message",
-      "user": "UC7SRSBH7",
-      "text": "PT-2 hahahahahaha db dfvbdffvdfvdfva ad asd",
-      "client_msg_id": "dad77307-b142-462b-85b6-e9f0b0a6dc34",
-      "edited": {
-        "user": "UC7SRSBH7",
-        "ts": "1546645882.000000"
-      },
-      "ts": "1546645707.005000"
-    },
-    "subtype": "message_changed",
-    "hidden": true,
-    "channel": "CF62ZFGFM",
-    "previous_message": {
-      "type": "message",
-      "user": "UC7SRSBH7",
-      "text": "PT-2 hahahahahaha db dfvbdffvdfvdfv",
-      "client_msg_id": "dad77307-b142-462b-85b6-e9f0b0a6dc34",
-      "edited": {
-        "user": "UC7SRSBH7",
-        "ts": "1546645740.000000"
-      },
-      "ts": "1546645707.005000"
+    "api_app_id": "A00T0R11P66",
+    "authorizations": [
+        {
+            "enterprise_id": null,
+            "is_bot": true,
+            "is_enterprise_install": false,
+            "team_id": "TT0EEPP4R",
+            "user_id": "U11PIPLPPSM"
+        }
+    ],
+    "event": {
+        "blocks": [
+            {
+                "block_id": "HeN",
+                "elements": [
+                    {
+                        "elements": [
+                            {
+                                "text": "hello",
+                                "type": "text"
+                            }
+                        ],
+                        "type": "rich_text_section"
+                    }
+                ],
+                "type": "rich_text"
+            }
+        ],
+        "channel": "C10P0P00303",
+        "channel_type": "group",
+        "client_msg_id": "fkj3b7s5-2w33-6n78-d33a-b7782s3v62s0",
+        "event_ts": "1638793296.000200",
+        "team": "TT0EEPP4R",
+        "text": "hello",
+        "ts": "1548324796.000200",
+        "type": "message",
+        "user": "U00TPRPIPPA"
```

Comment:
Note that, on top of a new message being posted to a channel, the previous version had captured the following variations/subtypes of the message event:
- When the bot joins a channel
- When the bot is removed from a channel
- When a message is edited/updated

Again, not a big deal but the point is that when handling Slack message events we have to be careful not to assume all events of this type are equal, but consider the subtypes.

Reply:
That was me who advised removing payload samples for other subtypes since they had the same structure.